### ISSUE TITLE
`init-pants` action does not respect `pants_bin_name` option

### ISSUE THREAD

**Issue by clydebw** [pantsbuild/actions#9]
I'm using the [`pants_bin_name`](https://www.pantsbuild.org/docs/reference-global#pants_bin_name) option to point Pants to `./pants-bin`. I've created my own `pants` script, which does two things:

- Refreshes a token to access our private CodeArtifact Python package index.
- Calls `pants-bin`, passing in all arguments that were provided to the `pants` script.

This allows us to keep CodeArtifact authenticated and use Pants in the standard way.

Right now I don't _think_ the `init-pants` GitHub Action respects `pants_bin_name` option, and [instead looks directly in `./pants`](https://github.com/pantsbuild/actions/blob/e63d2d0e3c339bdffbe5e51e7c39550e3bc527bb/init-pants/action.yaml#L40), which prevents us from using the latest version of this GitHub Action. (I might be wrong here, I'm pretty new to Pants)

Does it make sense to change the behaviour of this GitHub Action to check what `pants_bin_name` is (or perhaps the version of Pants), instead of having a look inside the `pants` script itself? If there is a recommended approach, I'm quite happy to attempt this change myself!

**Comment by benjyw** [comment]
Yep, looks like it uses `./pants` as a hardcoded name.

Grepping pants_bin_name might be complicated? Config can come from a few different places (and anyway, you're not required to set that, it's just for help messages, you can still rename `./pants` to whatever you like without setting that option). 

So I'd recommend adding an input to the job that sets the script name. We'd welcome a PR!

**Comment by cognifloyd** [comment]
Yeah, I think adding an input param to the action would be best here.
Probably `pants-bin-name` defaulting to `./pants`.

Note that the action actually introspects the `./pants` script to make sure it is new enough. So, if your `pants-bin-name` is a wrapper around the actual pants script, that might not work.

So, if you want to tackle this, please make sure to:
- replace all occurrences of `./pants` with something like `${{ inputs.pants-bin-name }}`
- make sure the pants script version check can still work somehow:

https://github.com/pantsbuild/actions/blob/615843b439af5072fce09f60362ab8a97482a8c0/init-pants/action.yaml#L41-L46

**Comment by cognifloyd** [comment]
Note: the `PANTS_BOOTSTRAP_TOOLS` check was introduced in #6. You can review the discussion and linked PRs there to see what the purpose of this check is.
